app.js

```diff
@@ -1,61 +1,94 @@
-
-server.set("view engine", "ejs");
-server.set("views", "./views");
-
-// handle iframe callback
-server.get("/callback", [urlEncodedMid], function (req, res, next) {
-    if (req.query.signature) {
-        if (validateSignature(req.query, configObj.iFrameSecret))
-            res.send("success signature");
-        else
-            res.send("not matched signature")
-    } else {
-        res.send("success signature");
-    }
-})
-
-server.get("/hppCallback", [urlEncodedMid], function (req, res, next) {
-    if (req.query.signature) {
-        if (validateSignature(req.query, configObj.HPPSecret))
-            res.send("success signature");
-        else
-            res.send("not matched signature")
-    } else {
-        res.send("success signature");
-    }
-})
-
-//render index.ejs
-server.get("/", [], function (req, res, next) {
-    let order = {
-        amount: 20.00,
-        currency: "EGP",
-        merchantOrderId: Date.now(),
-        mid: configObj.mid,
-        secret: configObj.iFrameSecret,
-        baseUrl: configObj.baseUrl,
-    }
-
-    // generate iframe hash
-    order.hash = generateKashierOrderHash(order);
-    order.secret = configObj.HPPSecret;
-
-    // generate HPP hash  
-    let phash = generateKashierOrderHash(order);
-    let callbackUrl = encodeURI('http://localhost:9000/hppCallback');
-
-    //Hosted payment page URL
-    let hppUrl = `${configObj.baseUrl}/payment?mid=${order.mid}&orderId=${order.merchantOrderId}&amount=${order.amount}&currency=${order.currency}&hash=${phash}&merchantRedirect=${callbackUrl}`
-    res.render('index', { order: order, hppUrl: hppUrl });
-})
+const bodyParser = require('body-parser');
+const path = require('path');
+
+server.set('view engine', 'ejs');
+server.set('views', './views');
+
+server.use(express.static(path.join(__dirname, 'kashier-files')));
+server.use(bodyParser.urlencoded({ extended: false }));
+server.use(bodyParser.json());
+
+server.use('/callback', require('./callBacks'));
+server.use('/PaymentWebhook', require('./PaymentWebhook'));
+
+//Order Configuration and index.js rendering.
+server.get('/', [], function (req, res, next) {
+  //Import merchant configuration
+  const config = require('./config');
+  const configObj = config[config.mode];
+
+  //Create your Order
+  let order = {
+    amount: '600.00',
+    // Add following option to specific order currency (ISO: "EGP", "USD", "GBP" "EUR")
+    currency: 'EGP',
+    // Unique order using as reference between merchant and kashier
+    merchantOrderId: Date.now(),
+    // Your Kashier Merchant ID 'MID-XXX-XX'
+    mid: configObj.mid,
+    // How to obtain your Payment API key
+    // Navigate to Integrate now page
+    // Click on Generate for Customizable form service
+    secret: configObj.PaymentApiKey,
+    // your website baseUrl, www.yourwebsite.com
+    baseUrl: configObj.baseUrl,
+    //order meta data JSON String
+    metaData: JSON.stringify({
+      'Product Name': 'Type Cable',
```

Comment:
Change the meta data to have more meanful options

Customer Name: "Mohamed Khaled"
Cutomer Phone : "+20100XXX"